ci-operator/step-registry/observers/resource-watch/observers-resource-watch-commands.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+export REPOSITORY_PATH="${ARTIFACT_DIR}/resource-watch-store"
+
+function cleanup() {
+  echo "killing resource watch"
+  CHILDREN=$(jobs -p)
+  if test -n "${CHILDREN}"
+  then
+    kill ${CHILDREN} && wait
+  fi
+
+  tar -czC $REPOSITORY_PATH -f "${ARTIFACT_DIR}/resource-watch-store.tar.gz" .
+  rm -rf $REPOSITORY_PATH
+
+  echo "ended resource watch gracefully"
+}
+trap cleanup SIGTERM
```

Comment:
May I suggest to replace this with `trap cleanup EXIT`?

Reply:
That worked! Thanks!